src/ranged.cpp

```diff
@@ -1209,11 +1221,17 @@ std::vector<tripoint> game::pl_target_ui( target_mode mode, item *relevant, int
                            m.mode.c_str(), m.qty );
             }
 
-            if( m->ammo_data() ) {
-                mvwprintw( w_target, line_number++, 1,
-                           m->ammo_capacity() > 1 ? _( "Ammo: %s (%d/%d)" ) : _( "Ammo: %s" ),
-                           item::nname( m->ammo_current(), m->ammo_remaining() ).c_str(),
-                           m->ammo_remaining(), m->ammo_capacity() );
+            auto cur = ammo ? ammo : m->ammo_data();
+            if( cur ) {
+                auto str = string_format( m->ammo_remaining() ?
```

Comment:
I see you're trying to save some space here but it's better to keep ```auto``` for some really long and complex iterator types.